sycl/include/CL/sycl/xilinx/fpga/opt_decorate_func.hpp

```diff
@@ -52,12 +52,11 @@ void dataflow(T functor) {
     \param[in] f is a function with an innermost loop to be executed in a
     pipeline way.
 */
-template <typename T>
-__SYCL_DEVICE_ANNOTATE("xilinx_pipeline") __attribute__((always_inline))
-void pipeline(T functor) {
+template<typename T>
+__SYCL_DEVICE_ANNOTATE("xilinx_pipeline")
+ALWAYS_INLINE void pipeline(T functor) {
```

Comment:
Why not use `[[gnu::always_inline]]` here? This would follow the standard C++ syntax and remove the need for using the preprocessor.

Reply:
I guess this is for handling any compiler, such as Microsoft Visual Studio?

Reply:
That is the beauty of this attribute syntax: compilers will ignore any attributes they don't understand. The worst that can happen is a compiler warning about using an unknown attribute.

Reply:
That is a good point. :-) Yes, the compiler might barf...

Reply:
Probably `ALWAYS_INLINE` is an unfortunate portable way to go.
I have the feeling it should have a more protected name, such as `__TRISYCL_ALWAYS_INLINE` or whatever.